### PR TITLE
Create aliases for ThreadPool methods to simplify threads usage

### LINKED ISSUE #38: Simplify thread system
Using the thread system should be really, really easy to encourage its use. The backend is probably fine as it is, but the user facing side should probably be simplified.  

The most straight forward solution would be just to use alias for the most common operations:

- [ ] `run_in_thread` function for `ThreadPool().run_thread`
- [ ] `abort_thread` function for `ThreadPool().abort_thread`
- [ ] `should_abort` function for `ThreadPool().query_abort`
- [ ] `run_daemon`, shortcut for running a function in a daemonic thread

### PR DESCRIPTION
**Pull request by srikanthravipati**
Includes aliases for https://github.com/ImperialCollegeLondon/python-gui-template/blob/8481970b163899b5febb58752077ae088c166bb5/pyguitemp/threads.py#L109 methods in line with tasks in #38

Closes #38

### CODE REVIEW

File: pyguitemp/threads.py [hunk 1]
```diff
@@ -185,3 +187,58 @@ def abort_thread(self, ident: int) -> None:
...
+
+    Args:
+        ident: Thread identifier
+
+    Raises:
+        KeyError: If the identifier is not in the ThreadPool
+    """
+    if ident in ThreadPool()._workers:
+        ThreadPool().abort_thread(ident)
+    else:
```
Comment (dalonsoa) [inline pyguitemp/threads.py:224]: This is a very good point, but I would not put it here. Instead, checking if `ident` is actually a thread managed by the `ThreadPool` should be done within `ThreadPool().abort_thread` such that both functions - the original and the alias - behave exactly the same.

File: pyguitemp/threads.py [hunk 2]
```diff
@@ -185,3 +187,58 @@ def abort_thread(self, ident: int) -> None:
...
+def should_abort(ident: int) -> bool:
+    """Return whether the thread with given identifier should abort or not.
+
+    Args:
+        ident: Thread identifier
+
+    Raises:
+        KeyError: If the identifier is not in the ThreadPool
+    """
+    if ident in ThreadPool()._workers:
```
Comment (dalonsoa) [inline pyguitemp/threads.py:239]: As above, this check should be done within `ThreadPool().query_abort`, so both functions behave in the same whay. This is particularly important in this case because `ThreadPool().query_abort` does not take any input argument while your implementation of `should_about` does.

File: tests/test_threads.py [hunk 3]
```diff
@@ -132,3 +132,56 @@ class WX:
...
+
+def test_run_daemon():
+    with patch("pyguitemp.threads.ThreadPool", MagicMock()):
+        from pyguitemp.threads import ThreadPool, run_daemon
+
+        run_daemon(lambda: None)
+        ThreadPool().run_daemon_called_once_with(None, None, None, None, True)
+
+
+def test_abort_thread():
```
Comment (dalonsoa) [inline tests/test_threads.py:157]: There should be two separate tests, one for `abort_thread` and another one for `should_abort`. If you find you need to repeat too much code, extract that as `fixtures`.

**Review by dalonsoa — CHANGES_REQUESTED** [review]
The original functions and the aliases should behave in exactly the same way, so I have suggested you move the checks you have implemented about `ident` - which are quite appropriate, by the way - into the original functions.

File: pyguitemp/threads.py (hunk 2 above)
Comment (srikanthravipati) [reply pyguitemp/threads.py:239]: That brings to me a question regarding the design choice. 

Choosing between passing `ident` explicitly or the existing implementation ` self._workers[threading.get_ident()].abort` that internally gets the identifier based on the thread that is calling.

 If the second option (existing implementation) is the preferred choice, how should we design the alias?. Also, this will be different to the `abort_thread` function that explicitly takes the `ident`.
 
There will be an explicit `ident` as an argument even in `query_abort` and will be similar to `should_abort`, in the case of the first choice. 

I am going ahead with the first choice, explicit `ident` in the changes. But depending on our final selection, I will revise it, if needed.

Comment (dalonsoa) [reply pyguitemp/threads.py:239]: The key point - which is not mentioned anywhere and should be explained - is that `abort_thread` is meant to be called by the main thread or any thread other than the one to be aborted, and so it needs as input the id of that thread. 

`query_abort`, on the contrary, is meant to be called _within the thread that, possibly, needs aborting_. In this case, the id of the thread can be atomaticlaly retrieved and does not need to be provided externally.

Threads, contrary to processes, cannot be killed (I'm talking about python here). The only way of finishing a thread is for the target function to return. `query_abort` is there so the target function can check from time to time if it needs aborting, and do so if required.

In summary: 
- `should_abort` must have `ident` as input.
- `query_abort` must have no inputs.

**Review by dalonsoa — APPROVED** [review]
